esmvalcore/preprocessor/_regrid.py

```diff
@@ -533,9 +535,13 @@ def _get_target_grid_cube(
     return target_grid_cube
 
 
-def _attempt_irregular_regridding(cube: Cube, scheme: str) -> bool:
+def _attempt_irregular_regridding(
+    src_cube: Cube,
+    tgt_cube: Cube,
+    scheme: str,
+) -> bool:
     """Check if irregular regridding with ESMF should be used."""
-    if not has_irregular_grid(cube):
+    if not (has_irregular_grid(src_cube) or has_irregular_grid(tgt_cube)):
```

Comment:
@schlunma Does this also work for ugrid data?

Reply:
Yes!

```py
>>> # cube is a UGRID-compliant cube
>>> ih.has_regular_grid(cube)
False
>>> ih.has_irregular_grid(cube)
False
>>> ih.has_unstructured_grid(cube)
True
```

Please note that in the UGRID word "irregular" and "unstructured" might mean the same, whereas for us "irregular" means "curvilinear"

Reply:
But for UGRID cubes we would like to use iris-esmf-regrid. What would be the best way to proceed? Add another function that checks if the source or target cube has a `iris.cube.Cube.mesh` property that is not `None` and that case use iris-esmf-regrid?

Reply:
Yeah, that sounds like a good suggestion to me!